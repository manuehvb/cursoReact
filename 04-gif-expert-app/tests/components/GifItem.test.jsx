import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";
import { screen } from "@testing-library/react";

describe('Tests de <GifItem />', () => {

    const tituloImagen = 'Titulo imagen';
    const urlImagen = 'http://localhost/urlImagen';

    test('GifItem Snapshot', () => {
        const { container } = render(<GifItem title={tituloImagen}
            url={urlImagen} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y ALT indicado', () => {
        render(<GifItem title={tituloImagen} url={urlImagen} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(urlImagen);
        expect(alt).toBe(tituloImagen);
    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={tituloImagen} url={urlImagen} />);
        expect(screen.getByText(tituloImagen)).toBeTruthy();
    });

});