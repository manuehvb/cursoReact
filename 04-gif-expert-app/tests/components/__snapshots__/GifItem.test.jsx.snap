// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests de <GifItem /> GifItem Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo imagen"
      src="http://localhost/urlImagen"
    />
    <p>
      Titulo imagen
    </p>
  </div>
</div>
`;
